esp-hal/src/gpio/mod.rs

```diff
@@ -364,7 +366,7 @@ pub trait RtcPin: Pin {
 /// Trait implemented by RTC pins which supporting internal pull-up / pull-down
 /// resistors.
 #[instability::unstable]
-#[cfg(any(soc_has_lp_io, soc_has_rtc_cntl))]
+#[cfg(not(esp32h2))]
```

Comment:
This is not great but we'll want to distance the LP IOMUX functionalities a bit anyway from HP GPIO. H2 doesn't have an LP mux/GPIO matrix, and P4 will have one pretty much equivalent in capabilities to the HP matrix, and we'll end up with a huge mess without separating the two.